tests/set/set4.c

```diff
@@ -35,9 +35,23 @@ int *next_int(void) {
 int main(void) {
 	struct set *s = set_create(NULL, cmp_int);
```

Comment:
Unchecked allocation.

Reply:
These are short test cases.  I can add an assert(), but ... if `set_create()` returns `NULL`, the test will fail with a segfault.